### ISSUE TITLE
Feature pop up not showing in leafletjs

### ISSUE THREAD

**Issue by brunogonzal** [tomchadwin/qgis2web#738]
Hi, 

When I export my map to Leaflet I cant get any popup to appear.
I think the problem is that  the function pop_Pernambuco_0(feature, layer) is empty but I don't know how I would write the code to make it work. Is there a model I could follow?

```
<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="initial-scale=1,user-scalable=no,maximum-scale=1,width=device-width">
        <meta name="mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <link rel="stylesheet" href="css/leaflet.css">
        <link rel="stylesheet" href="css/qgis2web.css"><link rel="stylesheet" href="css/fontawesome-all.min.css">
        <link rel="stylesheet" href="css/leaflet-measure.css">
        <style>
        #map {
            width: 976px;
            height: 822px;
        }
        </style>
        <title></title>
    </head>
    <body>
        <div id="map">
        </div>
        <script src="js/qgis2web_expressions.js"></script>
        <script src="js/leaflet.js"></script>
        <script src="js/leaflet.rotatedMarker.js"></script>
        <script src="js/leaflet.pattern.js"></script>
        <script src="js/leaflet-hash.js"></script>
        <script src="js/Autolinker.min.js"></script>
        <script src="js/rbush.min.js"></script>
        <script src="js/labelgun.min.js"></script>
        <script src="js/labels.js"></script>
        <script src="js/leaflet-measure.js"></script>
        <script src="data/Pernambuco_0.js"></script>
        <script>
        var map = L.map('map', {
            zoomControl:true, maxZoom:28, minZoom:1
        })
        var hash = new L.Hash(map);
        map.attributionControl.setPrefix('<a href="https://github.com/tomchadwin/qgis2web" target="_blank">qgis2web</a> &middot; <a href="http://leafletjs.com" title="A JS library for interactive maps">Leaflet</a>');
        var measureControl = new L.Control.Measure({
            position: 'topleft',
            primaryLengthUnit: 'meters',
            secondaryLengthUnit: 'kilometers',
            primaryAreaUnit: 'sqmeters',
            secondaryAreaUnit: 'hectares'
        });
        measureControl.addTo(map);
        document.getElementsByClassName('leaflet-control-measure-toggle')[0]
        .innerHTML = '';
        document.getElementsByClassName('leaflet-control-measure-toggle')[0]
        .className += ' fas fa-ruler';
        var bounds_group = new L.featureGroup([]);
        function setBounds() {
            if (bounds_group.getLayers().length) {
                map.fitBounds(bounds_group.getBounds());
            }
        }
        function pop_Pernambuco_0(feature, layer) {
        }

        function style_Pernambuco_0_0(feature) {
            switch(String(feature.properties['situacaoPE_Pista Liberada'])) {
                case 'A executar':
                    return {
                pane: 'pane_Pernambuco_0',
                opacity: 1,
                color: 'rgba(247,5,21,1.0)',
                dashArray: '',
                lineCap: 'square',
                lineJoin: 'bevel',
                weight: 6.0,
                fillOpacity: 0,
            }
                    break;
                case 'Executado':
                    return {
                pane: 'pane_Pernambuco_0',
                opacity: 1,
                color: 'rgba(37,97,217,1.0)',
                dashArray: '',
                lineCap: 'square',
                lineJoin: 'bevel',
                weight: 6.0,
                fillOpacity: 0,
            }
                    break;
                case 'Liberado':
                    return {
                pane: 'pane_Pernambuco_0',
                opacity: 1,
                color: 'rgba(123,165,81,1.0)',
                dashArray: '',
                lineCap: 'square',
                lineJoin: 'bevel',
                weight: 6.0,
                fillOpacity: 0,
            }
                    break;
            }
        }
        map.createPane('pane_Pernambuco_0');
        map.getPane('pane_Pernambuco_0').style.zIndex = 400;
        map.getPane('pane_Pernambuco_0').style['mix-blend-mode'] = 'normal';
        var layer_Pernambuco_0 = new L.geoJson(json_Pernambuco_0, {
            attribution: '',
            pane: 'pane_Pernambuco_0',
            onEachFeature: pop_Pernambuco_0,
            style: style_Pernambuco_0_0,
        });
        bounds_group.addLayer(layer_Pernambuco_0);
        map.addLayer(layer_Pernambuco_0);
        var baseMaps = {};
        L.control.layers(baseMaps,{' Pernambuco<br /><table><tr><td style="text-align: center;"><img src="legend/Pernambuco_0_Aexecutar0.png" /></td><td>A executar</td></tr><tr><td style="text-align: center;"><img src="legend/Pernambuco_0_Executado1.png" /></td><td>Executado</td></tr><tr><td style="text-align: center;"><img src="legend/Pernambuco_0_Liberado2.png" /></td><td>Liberado</td></tr></table>': layer_Pernambuco_0,}).addTo(map);
        setBounds();
        </script>
    </body>
</html>
```

**Comment by tomchadwin** [comment]
That popup function should not be empty - looks like a bug. Is there anything in the qgis2web log (View > Panels > Log Messages > qgis2web tab)?

**Comment by brunogonzal** [comment]
There is one message repeated 5 times:
2019-04-24T08:22:03     CRITICAL    Traceback (most recent call last):
              File "C:/Users/Bruno gonzalez/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\qgis2web\maindialog.py", line 360, in populate_layers_and_groups
              testDump = layer.renderer().dump()
             AttributeError: 'NoneType' object has no attribute 'dump'

**Comment by tomchadwin** [comment]
No, that's OK. I think we'll need to see your project and layers files.

**Comment by tomchadwin** [comment]
CLosing without feedback. If you upload your QGIS project and layer files, we can investigate further.

**Comment by brunogonzal** [comment]
Sorry, 
I was in a hurry with this project and then I forgot to reply. 
The problem was that the layers were in groups, and apparently qgis2web cant handle that well enough to create the popups. Once I removed the groups I could create the popups normally.
Thank you!

**Comment by tomchadwin** [comment]
Ah, glad you found the explanation, even if it a weakness in the plugin. For reference, the layer groups issue is here: https://github.com/tomchadwin/qgis2web/issues/175.

**Comment by andreaordonselli** [comment]
I have just released version [3.19.0](https://github.com/qgis2web/qgis2web/releases/tag/v3.19.0), which fixes the issues discussed in this thread. Shortly, the update will be available within the QGIS plugin repository.

Thank you for using the plugin.

I would greatly appreciate it if you would consider donating a virtual coffee through the button on the homepage to support the development of qgis2web.

Best regards.
